File: android/core/domain/src/main/java/com/enterprise/core/domain/entities/UIComponent.kt
package com.enterprise.core.domain.entities

import java.math.BigDecimal

// MARK: - Server-Driven UI Domain Entities

/** Polymorphic UI component model for server-driven rendering. */
sealed class UIComponent {
    abstract val id: String

    data class HeroCard(val config: HeroConfig) : UIComponent() {
        override val id: String get() = "hero-${config.id}"
    }

    data class ActionPillRow(val config: ActionPillConfig) : UIComponent() {
        override val id: String get() = "pills-${config.id}"
    }

    data class Carousel(val config: CarouselConfig) : UIComponent() {
        override val id: String get() = "carousel-${config.id}"
    }
}

/** Configuration for the full-bleed hero card. */
data class HeroConfig(
    val id: String,
    val title: String,
    val subtitle: String,
    val imageUrl: String?,
    val gradientColors: List<String>,
    val pricing: PricingInfo,
    val ctaText: String,
    val ctaDeepLink: String
)

/** Pricing information for product cards. */
data class PricingInfo(
    val amount: BigDecimal,
    val currency: String,
    val period: String,
    val originalAmount: BigDecimal? = null
)

/** Configuration for action pill row (Bill, Usage, etc.). */
data class ActionPillConfig(
    val id: String,
    val pills: List<ActionPill>
)

data class ActionPill(
    val label: String,
    val iconName: String,
    val deepLink: String
)

/** Configuration for horizontal carousel (Just For You bundles). */
data class CarouselConfig(
    val id: String,
    val title: String,
    val items: List<CarouselItem>
)

data class CarouselItem(
    val id: String,
    val title: String,
    val imageUrl: String?,
    val price: PricingInfo?,
    val deepLink: String
)

File: ios/EnterpriseMobileApp/Sources/Core/Domain/Entities/UIComponent.swift
import Foundation

// MARK: - Server-Driven UI Domain Entities

/// Polymorphic UI component model for server-driven rendering.
public enum UIComponent: Equatable, Sendable, Identifiable {
    case heroCard(HeroConfig)
    case actionPillRow(ActionPillConfig)
    case carousel(CarouselConfig)

    public var id: String {
        switch self {
        case .heroCard(let config): return "hero-\(config.id)"
        case .actionPillRow(let config): return "pills-\(config.id)"
        case .carousel(let config): return "carousel-\(config.id)"
        }
    }
}

/// Configuration for the full-bleed hero card.
public struct HeroConfig: Equatable, Sendable {
    public let id: String
    public let title: String
    public let subtitle: String
    public let imageURL: URL?
    public let gradientColors: [String]
    public let pricing: PricingInfo
    public let ctaText: String
    public let ctaDeepLink: String

    public init(
        id: String,
        title: String,
        subtitle: String,
        imageURL: URL?,
        gradientColors: [String],
        pricing: PricingInfo,
        ctaText: String,
        ctaDeepLink: String
    ) {
        self.id = id
        self.title = title
        self.subtitle = subtitle
        self.imageURL = imageURL
        self.gradientColors = gradientColors
        self.pricing = pricing
        self.ctaText = ctaText
        self.ctaDeepLink = ctaDeepLink
    }
}

/// Pricing information for product cards.
public struct PricingInfo: Equatable, Sendable {
    public let amount: Decimal
    public let currency: String
    public let period: String
    public let originalAmount: Decimal?

    public init(amount: Decimal, currency: String, period: String, originalAmount: Decimal? = nil) {
        self.amount = amount
        self.currency = currency
        self.period = period
        self.originalAmount = originalAmount
    }
}

/// Configuration for action pill row (Bill, Usage, etc.).
public struct ActionPillConfig: Equatable, Sendable {
    public let id: String
    public let pills: [ActionPill]

    public init(id: String, pills: [ActionPill]) {
        self.id = id
        self.pills = pills
    }
}

public struct ActionPill: Equatable, Sendable {
    public let label: String
    public let iconName: String
    public let deepLink: String

    public init(label: String, iconName: String, deepLink: String) {
        self.label = label
        self.iconName = iconName
        self.deepLink = deepLink
    }
}

/// Configuration for horizontal carousel (Just For You bundles).
public struct CarouselConfig: Equatable, Sendable {
    public let id: String
    public let title: String
    public let items: [CarouselItem]

    public init(id: String, title: String, items: [CarouselItem]) {
        self.id = id
        self.title = title
        self.items = items
    }
}

public struct CarouselItem: Equatable, Sendable, Identifiable {
    public let id: String
    public let title: String
    public let imageURL: URL?
    public let price: PricingInfo?
    public let deepLink: String

    public init(id: String, title: String, imageURL: URL?, price: PricingInfo?, deepLink: String) {
        self.id = id
        self.title = title
        self.imageURL = imageURL
        self.price = price
        self.deepLink = deepLink
    }
}
